### ISSUE TITLE
[REVIEW]: NeuralFieldEq.jl: A flexible solver to compute Neural Field Equations in several scenarios

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3974]
**Submitting author:** @tiagoseq (<a href="http://orcid.org/0000-0001-8579-3676">Tiago Sequeira</a>)
**Repository:** https://github.com/tiagoseq/NeuralFieldEq.jl
**Branch with paper.md** (empty if default branch): 
**Version:** v0.1.3
**Editor:** @jarvist
**Reviewers:** @rougier, @gdetor
**Archive:** 10.5281/zenodo.6226695

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/85354f0f2b6244439e90d51fe28a9ef5/status.svg)](https://joss.theoj.org/papers/85354f0f2b6244439e90d51fe28a9ef5)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/85354f0f2b6244439e90d51fe28a9ef5"><img src="https://joss.theoj.org/papers/85354f0f2b6244439e90d51fe28a9ef5/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/85354f0f2b6244439e90d51fe28a9ef5/status.svg)](https://joss.theoj.org/papers/85354f0f2b6244439e90d51fe28a9ef5)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/tiagoseq/NeuralFieldEq.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@rougier & @gdetor, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @jarvist know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @rougier
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/tiagoseq/NeuralFieldEq.jl">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/tiagoseq/NeuralFieldEq.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@tiagoseq) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @gdetor
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/tiagoseq/NeuralFieldEq.jl">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/tiagoseq/NeuralFieldEq.jl">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@tiagoseq) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @rougier, @gdetor it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 1046

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.16 s (251.7 files/s, 117203.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
CSS                              2            701             49          14825
Julia                           14            190            158            845
Markdown                         7            141              0            538
JavaScript                       5             44             78            511
HTML                             5             34              0            233
TeX                              1             13              0            121
YAML                             3              0              1            114
TOML                             2              5              0             20
JSON                             1              0              0              7
-------------------------------------------------------------------------------
SUM:                            40           1128            286          17214
-------------------------------------------------------------------------------


Statistical information for the repository '020d7a1387adc6bd5fcc8570' was
gathered on 2021/12/03.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
tiagoseq                         1           633              0          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
tiagoseq                    633          100.0          0.0               12.32
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1113/jphysiol.1952.sp004764 is OK
- 10.1109/JPROC.2004.840301 is OK
- 10.1002/cne.21974 is OK
- 10.1016/S0006-3495(72)86068-5 is OK
- 10.1103/PhysRevE.82.055701 is OK
- 10.1007/978-3-642-54593-1_6 is OK
- 10.3389/fninf.2015.00025 is OK
- 10.1186/2190-8567-4-1 is OK
- 10.1007/BF00337259 is OK
- 10.1137/141000671 is OK
- 10.1109/ICSTCC.2019.8885972 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by jarvist** [comment]
Dear @rougier and @gdetor, how are things coming along?
I think your invitation link may have expired @gdetor.

**Comment by jarvist** [comment]
@whedon re-invite @gdetor as reviewer

Just in case!

**Comment by whedon** [comment]
The reviewer already has a pending invite.

@gdetor please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by jarvist** [comment]
OK! Thanks @whedon.

**Comment by whedon** [comment]
:wave: @gdetor, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @rougier, please update us on how your review is going (this is an automated reminder).

**Comment by rougier** [comment]
Here are some temporary comments on the submission since I'm waiting for an [issue](https://github.com/tiagoseq/NeuralFieldEq.jl/issues/29) to be solved before going further.

1. Repository is way too big (36Mo) due mostly to the presence of gif animations in the imag and doc directories. Problem is that the doc directories containes the build website for no obvious reason. I would advise to cleanup the repository. But problem will remain because the build addition has been commited and the repository will stay big unless the git history is rewritten. Another solution would be to create a new repository and make sure to only add necessary stuff.

2. The name of the repository is a bit weird with the .jl extension but I don't know if it is usage for Julia packages.

3. Since computation is a bit long, would it be possible to have some kind of progress indicator when it is ran?

4. The summary in the paper needs to be worked a bit. Especially the first sentence that seems to be missing a verb. But more generally, I think i would be ok to dive directly in the neural field, starting at "Wilson & Cowan (1972)...".

5. Equation 1 seems to be missing the axonal speed term. This is ok since it does not appear in the original equation but you shoudl rephrase the paragraph to explain that speed can be introduced in the equation and then rewrite the full equation.
   
5. "when the coupling is positive (negative) the synapses are excitatory (inhibitory)" -> "when the coupling is positive (resp. negative), the synapses are excitatory (resp. inhibitory)

6. "Although the this" -> "Although this"

7. Overall, I think the paper could be improved because there are some typos that can easily be fixed and the package usage section should be improved to better introduce the package. As it is currently written it's a bit hard to follow author's explanations.

**Comment by tiagoseq** [comment]
Hello,

I can answer some of the comments:

1. Indeed the repo is too big due to the animations present in the docs section. I'll see what I can do to solve this problem, I guess I will take your advice and create a new repository for the docs, or re-write "lighter" documentation without too many animations.
I have to address this question carefully, I'll give updates on this soon.

2. The name of the repository follows the julia guidelines about package names, that's why the extension .jl appears. For example, these packages have the extension: [DifferentialEquations.jl](https://github.com/SciML/DifferentialEquations.jl) and [FFTW.jl](https://github.com/JuliaMath/FFTW.jl).

3. I guess that is possible to add a progress indicator, I'll give updates soon.

4. - 7. Since these issues are easier to fix, I'll check them first. I agree with you, the summary needs to be written more carefully.

8. Ok, I will read carefully again the package usage section and try to improve it to make it more clear.

When I have some of these points done, I'll write here.

Thanks,
Tiago

**Comment by gdetor** [comment]
Here are my comments on the manuscript. I'll provide my comments on the source code later. 

Text Comments
-----------------

1. Memory, perception, and other functions of the human brain do not occur only in the cerebral cortex. Please rephrase the sentences in lines 7 - 10. 

2. It would be better to rewrite equation 1 in a more general way. For instance, the kernel function K(x, t) is not always isotropic or homogeneous. You could incorporate the delay term as well. Why is the norm a Euclidean one (is the only supported norm)? The integral in equation 1 is always two-dimensional (please replace the square with d in the differential of the integral). You could also use a different symbol for indicating the dimension. What about 3-dimensional neural fields? 

3. You could consider citing these two references: Coombes, "Waves, bumps, and patterns in neural field theories", 2005 and Bressloff, "Spatiotemporal dynamics of continuum neural fields", 2011. Thus, readers who are not experts in neural fields can get a general idea. 

4. Could you please provide a reference regarding the performance of Julia? See lines 40-42 in the main text. 

5. The first paragraph in the "Package usage" section is obscure. This section needs some improvements. 

6. The sentence in lines 44-46 is not clear. 

7. Could you provide more details regarding the performance of the code? 

Language
----------
The text needs polishing. Here are some typos I spotted in the main text: 
1. Please remove "that" from the very first sentence of the summary. 
2. Line 30, "Although the this ..." -> "Although this ..."
3. Lines 35-36, "the scope of the master's thesis [Ref] ..." -> "the scope of [Ref]"
4. Line 42, "the the" -> "the"
5. Line 44, "non-delayed or delayed equations" -> "non-delayed or delayed"
6. Line 58, "access it at the previously" -> "access it at previously"

**Comment by jarvist** [comment]
Dear @tiagoseq , 
How is it coming along?

**Comment by tiagoseq** [comment]
Hello,
Sorry for the delay, unfortunately, only yesterday was possible for me to address the comments pointed out. 
I'm planning to reply to it during this week.

I'll give updates soon

**Comment by tiagoseq** [comment]
Hello,
I've released a new major version of the manuscript, with the following changes:
- Reformulated the Package usage section, i think now is more straight to the point. (both reviewers suggestion @rougier and @gdetor);
- The code example had some modifications;
- Introduced the delay term in the equation (both reviewers suggestion), as well as the delayed stochastic NFE;
- Added more context and references to the Neural Field Equations (@gdetor suggestion) in Summary section;
- Introduced new paragraph with two new references in Statement of need section, to give more context about the importance of rapid and efficient solvers; 
- Opened a new section called Example and code performance, and moved the example to this section and added a table with benchmarks for the 1D and 2D problems for several spatial nodes;
- Polished the text and corrected the typos.

Now, answering some questions:
@rougier 
1. Regarding the repo size, i've checked the settings of the repo on github and says that is about 10 Mb. Am I seeing this wrong? Yet, I have to work in this issue;
2.  I thought that add a status bar would be straightforward, but it isn't. Probably, next week I have to check it out.

@gdetor 
 1. I wrote an isotropic kernel because the solver do not support (yet) such functions. In this method K is homogeneous and isotropic (check line 15);
 2. Yes, the euclidean norm is the only supported, the method implemented only considers p-2 norms;
 3. I did not addressed the 3D case, because the first motivation of the solver was to study fields that exhibit working memory mechanisms applied to cognitive robotics (first paragraph of Statement of need) and there was no need to have one more dimension.

Thank you, and I'll be grateful when you can give feedback,
Tiago

**Comment by rougier** [comment]
@tiagoseq 

You can check the size locally with command `du -sh NeuralFieldEq.jl` (Linux or OSX). I think size reported by GitHub is only for current files, without the history of commits which can be quite large.

For the progress bar, it's only a suggestion, nothing really important. There is certainly a Julia package to do just that. If it's the case that would be nice and easy to add it.

**Comment by gdetor** [comment]
@tiagoseq @jarvist 
Here are my comments on the Source Code: 
1. Many parts of the code are not documented (for instance the functions in script AuxScripts.jl or in SolveStruct.jl). Please document all the functions and variables within the code. 
2. Please provide a list with the contributors (on the Github repository as a section in the README file)
3. Please add community guidelines (especially for issues and bugs) in the repo's README file. 

Other than that everything looks fine to me. The installation instructions are OK. The code works properly. The performance of the package is now reported in the main text. The repository contains automated tests and all tests pass with no any apparent issues. 

My previous comments regarding the main text have been addressed by the author. 

@tiagoseq Could you please generate the new pdf article proof?

**Comment by tiagoseq** [comment]
@gdetor
- During this week I'll provide the documentation for the auxiliary functions and the functors present in the SolveStruct.jl and others that might need more insights and explanation.
- Regarding the contributors to the package, (until now) I was the one responsible for writing it. But I'll specify that in the README file as you suggested.
- Yes, I'll write the community guidelines.

The new version of the manuscript is generated.

@rougier 
I will move the docs to an independent repository and clean the git history to reduce the size of the repo. I'll look this issue after wednesday. Thank you for this suggestion!
Yes, there is a Julia package to implement a status bar, probably is straight forward. I like that nice-to-have feature, I'll try to implement it after addressing the major issues pointed out!

Thank you!

**Comment by tiagoseq** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03974/joss.03974/10.21105.joss.03974.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03974/joss.03974/10.21105.joss.03974.pdf) :page_facing_up: :point_left:

**Comment by rougier** [comment]
@tiagoseq PDF has been rebuilt, does that mean you adressed our comments  with @gdetor?

**Comment by tiagoseq** [comment]
Hello,
Sorry for the delay, today I finally had time to work on this issue.
@rougier yes, I rebuilt the PDF with the comments that you and @gdetor made to the manuscript. Every comment, suggestion was addressed in this new version.

As for the code, what is already done is:
- I added the status when executing the function `solveNFE` @rougier 
- Updated readme with a new section that contains the contributors, and community guidelines @gdetor 
- Documented structures `ProbStruct.jl` and `SolveStruct.jl`, the module `AuxFunctions.jl` is documented and `DomainConstruct.jl`.

What I'll do next @rougier :
- Move documentation to other independent repository
- Clean git commit history
- Release a new version of the software.

When the new version is released, I'll communicate here

**Comment by tiagoseq** [comment]
Hello,
A pull request with a new version with the following features has been made:
- Status bar when executing function `solveNFE`
- New independent repository, [NFEdocs.jl](https://tiagoseq.github.io/NFEdocs.jl/dev/) to host the documentation of the package
- Documented structures and the auxiliary functions module
- Cleaned git commit history by creating an orphan branch with the latest commit and files, deleting the master branch and renaming the orphan branch to master. (However, I have to check this after the auto-merge)

I'm waiting for the [auto-merge](https://github.com/JuliaRegistries/General/pull/55127), made by JuliaRegistrator autobot. When the new version (v0.1.7) is released I'll let you know.

**Comment by rougier** [comment]
Any progress?

**Comment by tiagoseq** [comment]
Hello,
Yes, the new version of the package has been released. To update, please, go to julia terminal and type `] update`. One can see the new documentation and the status bar, while executing the function `solveNFE`.

However, the package is still too big. Even after moving the documentation to an independent repository, and after the creation of a new orphan branch and deleting the old master branch and commit history.

I have to check this question carefully. The problem relies on `.git` folder. I've tried another approaches but without success. Probably tomorrow I'll give some updates.

Nevertheless, the new version of the package can be checked after updating the local julia registries. If you have any issue updating the version 0.1.3 tell me.

Thank you!

**Comment by rougier** [comment]
Thanks for the update. For the size of the repository, I think it's ok. The .git folder probably carry the whole history and this might explain the size. For future code, make sure to not add all the files blindly or the problem will reappear. I've checked the new PDF and it now looks good to me. @jarvist @gdetor Looks good to me.

**Comment by tiagoseq** [comment]
Yes, learned my lesson. However, I still want to overcome this issue, to future downloads not to be so heavy.
Thank you for your advice to add a progress bar @rougier. 
If i have any update about the commit history I'll let you know.

**Comment by gdetor** [comment]
@tiagoseq Thank you for the update. 
@jarvist @rougier Both the manuscript and the source code repository look good to me. All my comments have been addressed.

**Comment by tiagoseq** [comment]
Hi @gdetor thank you for your help in improvig the manuscript and the source code!

In order to move on with this process, I think you must complete your check list, as @rougier did, at the top of this thread.

Thanks,
Tiago

**Comment by gdetor** [comment]
@jarvist @tiagoseq It seems I cannot access the check list. Any help?

**Comment by tiagoseq** [comment]
I searched for an answer in [JOSS readthedocs](https://joss.readthedocs.io/en/latest/index.html) and didn't see anything with respect on how complete the checklist.

@jarvist can you give any help? Or maybe @rougier could give some insight on how complete the check list

**Comment by tiagoseq** [comment]
Hello @jarvist, any help to @gdetor on the issue of filling out the task list? Just like @rougier already did

Thanks!

**Comment by tiagoseq** [comment]
> @jarvist @tiagoseq It seems I cannot access the check list. Any help?

Hi @gdetor , just to check, when you say you cannot access the check list in the beginning of this thread, you're saying that you could not edit the comment that contains the checklist, right?
It seems that @rougier didn't have any problem with this:
![image](https://user-images.githubusercontent.com/73851660/159727697-660e8c19-1fed-43e7-ba76-fcc2f8438ae8.png)

I can't edit the edit the check list either, because probably only the reviewers can.
When reading the initial comments, I found the following post:

> The reviewer already has a pending invite.
> 
> @gdetor please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

Did you accepted this invitation? If not, probably this is the problem. If you accepted the invitation to be a reviewer then I guess only @jarvist can help us.

Thank you

**Comment by gdetor** [comment]
@tiagoseq That invitation has expired. So that might be the problem.  
@jarvist Could you please send a new invitation? Thanks

**Comment by danielskatz** [comment]
bot commands need to be the first thing in a comment, but also, `@whedon` has been replaced by `@editorialbot`, and our method for generating and using checklists has changed.  In this case, and for new reviews going forward, the reviewer (@gdetor) should put `@editorialbot generate my checklist` at the start of a new comment to get a checklist to use. This is a bit confusing here, since this review started under the old system.

**Comment by gdetor** [comment]
@editorialbot generate my checklist

**Comment by tiagoseq** [comment]
Hi @gdetor, did you succeed in completing the checklist?

@danielskatz Thank you for your help! So, if the review started under the old system, what does it means in practice? That the other reviewer, @rougier, has to generate a new checklist with the @editorialbot ? Or it is ok to proceed the process with one reviewer with the "old" checklist and the other one with the checklist generated with the @editorialbot ?

**Comment by danielskatz** [comment]
This can stay as is, with mixed checklists.

@gdetor might want to try the command again, as GitHub was dropping some requests for a bit yesterday.

**Comment by gdetor** [comment]
@editorialbot generate my checklist

**Comment by gdetor** [comment]
@danielskatz Thank you for the instructions. 
@tiagoseq Thank you, now it should be OK.

**Comment by tiagoseq** [comment]
Thank you @gdetor !

@danielskatz , now that both checklists are completed, what is the next step of the process? Is there anything I can do?

**Comment by danielskatz** [comment]
@jarvist is the editor, so this should be directed to him.

**Comment by tiagoseq** [comment]
Hello,
It seems @jarvist is absent, does not participate in this thread for a long time.

@danielskatz is there anything i can do in this case?

**Comment by jarvist** [comment]
Sorry for my absence! It was a busy end of term anyway, and then I was ill & off work for a while.

**Comment by jarvist** [comment]
@editorialbot generate pdf

**Comment by jarvist** [comment]
@editorialbot check references

**Comment by jarvist** [comment]
@tiagoseq - I must admit I'm more than a little bit concerned at the fact that you appear to have deleted all git history & then force-pushed to your GitHub repository, since the start of this review. Was this intentional? Could you explain why?

I was just trying to check that it made sense for you to be the sole author on the paper, c.f. the contribution to the package, and the fact that you have your (academic?) supervisor in the acknowledgments only.

**Comment by tiagoseq** [comment]
Hello @jarvist , yes, it was intentional. I was trying to address the issue that @rougier pointed out, regarding the repository size,
>     1. Repository is way too big (36Mo) due mostly to the presence of gif animations in the imag and doc directories. Problem is that the doc directories containes the build website for no obvious reason. I would advise to cleanup the repository. But problem will remain because the build addition has been commited and the repository will stay big unless the git history is rewritten. Another solution would be to create a new repository and make sure to only add necessary stuff.

As Rougier said, the large size was due to the gifs and animations that are present in the documentation.
To resolve this I created a new repository for the sole purpose of hosting the package documentation, [NFEdocs.jl](https://tiagoseq.github.io/NFEdocs.jl/dev/), then I deleted the commit history in an attempt of reducing the size of the repo.

On the question of being the only author of the paper, I wrote the entire julia package from scratch, since this is integrated in the scope of my master thesis I did not receive any input from other students or professors. Moreover, the entire manuscript was written by me and my academic supervisor read the manuscript and pointed out some typos, before I submitted it.

I hope the questions have been clarified!

Thank you,
Tiago

**Comment by jarvist** [comment]
> To resolve this I created a new repository for the sole purpose of hosting the package documentation, [NFEdocs.jl](https://tiagoseq.github.io/NFEdocs.jl/dev/), then I deleted the commit history in an attempt of reducing the size of the repo.

Ah, OK! I understand the motivation now. But this is quite sub optimal, as it stops a future user / maintainer being able to `git bisect` or `git blame` while debugging, and otherwise understanding the construction of the package. 
Presuming you still have the original repository, rewriting the history to remove the large files & then force-pushing the resulting repo to github would be ideal. 
This kind of git stuff is a little bit scary & difficult. I think I used `git-filter-repo` the last time I needed to do this: https://htmlpreview.github.io/?https://github.com/newren/git-filter-repo/blob/docs/html/git-filter-repo.html

**Comment by rougier** [comment]
@jarvist I confirm that I advised @tiagoseq to create a new repo but you're right that it might be better to remove the faulty commit from the original repo.

**Comment by jarvist** [comment]
Dear @tiagoseq , have you been able to make any progress in retrieving some of the commit history?

**Comment by tiagoseq** [comment]
Hello, i'm very sorry for the late reply.
Unfortunately, about 2 months ago i had a car accident, nothing too serious.
The major injuries that i had were a broken leg and a total rupture of my ACL. 
And since then I have undergone two surgeries (one of them performed last week) and lots of physiotherapy. 

I simply didn't have time yet to solve this issue.

Again, i'm really sorry and i'll try to address this issue as soon as possible

**Comment by jarvist** [comment]
Dear Tiago,
Very sorry to hear that! Hope you mend well. Of course we can leave this open for the time being. I'll check in again in 4 weeks to see where we stand. 
Best wishes!

**Comment by jarvist** [comment]
Ok, I'll try and remind myself :^)

**Comment by rougier** [comment]
Or ask the bot to remind me and I'll remind you.

**Comment by tiagoseq** [comment]
Hello,
After a few weeks of a hard physical recovery I'm finally able to address this issue.
First of all, I want to thank you all, @jarvist, @rougier and @gdetor for the patience shown throughout this issue.

Regarding the problem that @jarvist pointed out, possible future contributors could not do `git blame` or `git bisect` because the history has been erased. An attempt to reduce the size of the repo, due to the number of GIFs used in the documentation.

However, the only major change that the current version of the repository has in comparison to the previous one (the one that I deleted the commit history) is the docs are hosted in an independent [repository](https://github.com/tiagoseq/NFEdocs.jl), the other changes are minor, such as the addition of a status bar and clean up the function documentation.
The solver itself is pretty much the same as it was in the first released version, v.0.1.0.

Basically, the current version of the solver is the baseline and I (and future contributors) will add new features, new numerical methods and schemes, etc on top of this version of the code.
Actually, I will add a road map to the readme section to point out what are the next steps.

Since these `git` things are a bit scary for me and the difference between v.01.0 and the current v0.1.3, is basically the docs hosted in a different repo and a status bar, I think that for future contributors there will be no problem to trace back possible bugs and errors. The very core computations of the solver weren't changed since its first release.


Once again, thank you so much for your patience.
I'm waiting for your response!
Tiago

**Comment by jarvist** [comment]
OK! Given the mitigating circumstances, and having read the code (and seen that it well commented and has good test coverage), I think this is acceptable. 

I've just made a few suggested edits to your paper, for stylistic reasons: 
https://github.com/tiagoseq/NeuralFieldEq.jl/pull/32

**Comment by tiagoseq** [comment]
Thank you @jarvist. I've already seen the suggestions and accepted the pull request and merged!

**Comment by jarvist** [comment]
@editorialbot check references

**Comment by jarvist** [comment]
@tiagoseq - could you generate a DOI of your v0.1.3 tag? Then we'll all set for recommending acceptance.

**Comment by tiagoseq** [comment]
I've already generated a DOI for this version (using Zenodo): [DOI for v.0.1.3](https://doi.org/10.5281/zenodo.6226695)

**Comment by jarvist** [comment]
@editorialbot set 10.5281/zenodo.6226695 as archive

**Comment by jarvist** [comment]
@editorialbot recommend-accept

Congratulations @tiagoseq ! An editor in chief will now take over to double check the proofs. 🚀

**Comment by jarvist** [comment]
Ah! I didn't notice the missing DOI earlier. Ideally that would be added if possible @tiagoseq .

**Comment by tiagoseq** [comment]
Done, the missing DOI is already inserted (and the correspondent .bib entry updated). Do I need to generate the paper with the new entry?

**Comment by jarvist** [comment]
@editorialbot generate pdf

**Comment by jarvist** [comment]
@editorialbot check references

**Comment by tiagoseq** [comment]
@editorialbot generate pdf

**Comment by tiagoseq** [comment]
@editorialbot check references

**Comment by tiagoseq** [comment]
Hi @jarvist! I do not know why the DOI is still invalid, I've already updated the respective entry. I've copied the bib entry from the journal's website. The paper was published in June of this year, maybe this is the 'problem'?

**Comment by arfon** [comment]
@editorialbot recommend-accept

**Comment by arfon** [comment]
@editorialbot accept

**Comment by arfon** [comment]
@rougier, @gdetor – many thanks for your reviews here and to @jarvist for editing this submission! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@tiagoseq – your paper is now accepted and published in JOSS :zap::rocket::boom:
